pkg/plugins/manager/loader/angulardetector/gcom.go

```diff
@@ -0,0 +1,204 @@
+
+	// gcomAngularPatternsPath is the relative path to the GCOM API handler that returns angular detection patterns.
+	gcomAngularPatternsPath = "/api/plugins/angular_patterns"
+)
+
+var _ detectorsProvider = &gcomDetectorsProvider{}
+
+// gcomDetectorsProvider is a detectorsProvider which fetches patterns from GCOM, and caches the result for
+// the specified ttl. All subsequent calls to provideDetectors will return the cached result until the TTL expires.
+// This struct is safe for concurrent use.
+type gcomDetectorsProvider struct {
+	log log.Logger
+
+	httpClient *http.Client
+
+	baseURL string
+
+	ttl        time.Duration
+	lastUpdate time.Time
+
+	mux       sync.Mutex
+	detectors []detector
+}
+
+// newGCOMDetectorsProvider returns a new gcomDetectorsProvider.
+// baseURL is the GCOM base url, without /api and without a trailing slash (e.g.: https://grafana.com)
+// A default reasonable value for ttl is defaultGCOMDetectorsProviderTTL.
+func newGCOMDetectorsProvider(baseURL string, ttl time.Duration) (detectorsProvider, error) {
+	cl, err := httpclient.New()
+	if err != nil {
+		return nil, fmt.Errorf("httpclient new: %w", err)
+	}
+	return &gcomDetectorsProvider{
+		log:        log.New("plugins.angulardetector.gcom"),
+		baseURL:    baseURL,
+		httpClient: cl,
+		ttl:        ttl,
+	}, nil
+}
+
+// tryUpdateRemoteDetectors tries to update the cached detectors value, if the cache has expired.
+//
+// If the TTL hasn't passed yet, this function returns immediately.
+// Otherwise, it calls fetch and updates the cached detectors value and lastUpdate.
+//
+// lastUpdate is also updated in case of an error, to avoid consecutive failures.
+// However, if there's an error, the cached value is not changed (the previous one is kept).
+//
+// The caller must have acquired g.mux.
+func (p *gcomDetectorsProvider) tryUpdateRemoteDetectors(ctx context.Context) error {
+	if time.Since(p.lastUpdate) <= p.ttl {
+		// Patterns already fetched
+		return nil
+	}
+
+	// Update last update even if there's an error, to avoid wasting time due to consecutive failures
+	defer func() {
+		p.lastUpdate = time.Now()
+	}()
+
+	// Fetch patterns from GCOM API
+	resp, err := p.fetch(ctx)
+	if err != nil {
+		return fmt.Errorf("fetch: %w", err)
+	}
+
+	// Convert patterns definitions to detectors
+	detectors, err := resp.detectors()
+	if err != nil {
+		return fmt.Errorf("detectors: %w", err)
+	}
+	p.log.Debug("Updated remote angular detectors", "detectors", len(detectors))
+
+	// Update cached result
+	p.detectors = detectors
+	return nil
+}
+
+// provideDetectors gets the remote detections, either from the cache or from the remote source (if TTL has passed).
+// If an error occurs during the cache refresh, the function fails silently and the old cached value is returned
+// instead.
+func (p *gcomDetectorsProvider) provideDetectors(ctx context.Context) []detector {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	if err := p.tryUpdateRemoteDetectors(ctx); err != nil {
```

Comment:
If I am reading this correctly, this will block the startup process until the detectors are fetched (if the entries are empty or no longer valid). We should try to avoid that if possible. The approach we followed for the public manifest was to make this a background service, and eventually update the value (in this case, the detectors). In the first run, the dynamic detector would return the hardcoded detectors but in the next server start, it will use the cached values that were fetched asynchronously. 

Note that you need to use persistent storage for that so look at how that's done for the public manifest.

Reply:
Correct, I did it this way as I didn't want to use the storage, but I can refactor it so they are similar and use the storage like for the signing keys

Reply:
I will remove caching from this PR as suggested, so resolving this